hashintel/hash
# Implement blurDataURL for hashdotdev blog post list and blog content

## 🌟 What is the purpose of this PR?

This PR implements the blurred preview images for the blog images to provide a better UX and prevent layout shifting / looking blank on slower connections 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203623179643290/1204086167143735/f) _(internal)_


## 🔍 What does this change?

-  `getPhoto` util function now adds `blurDataURL` to the data using `plaiceholder`
- Added a new mdx plugin to `rehypePlugins` called `imageMetadata`, which parses the `.mdx` file, filters all the images, then adds `blurDataURL` attribute to them

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Enable throttling (e.g. Fast 3G) to test slower internet
1. Confirm that images at `/blog` and `/blog/[blog-post-id]` has blurred previews

## 📹 Demo

https://user-images.githubusercontent.com/39553853/224286685-a9426416-b974-4924-aa9b-f33c4f17efb5.mov
